packages/core/src/shared/utilities/collectionUtils.ts

```diff
@@ -329,6 +330,33 @@ export function partialClone(obj: any, depth: number = 3, omitKeys: string[] = [
     return clonedObj
 }
 
+type inspectOptions = Partial<{
+    depth: number
+    omitKeys: string[]
+    replacement: any
+    showHidden: boolean
+    color: boolean
+}>
+
+/**
+ * Wrapper around nodes inspect function that works on web. Defaults to JSON.stringify on web.
+ * @param obj object to show
+ * @param opt options for showing (ex. depth, omitting keys)
+ */
+export function formatObj(obj: any, opt?: inspectOptions): string {
+    const options = {
+        depth: opt?.depth ?? 3,
+        omitKeys: opt?.omitKeys ?? [],
+        replacement: opt?.replacement,
+        showHidden: opt?.showHidden ?? false,
+        color: opt?.color ?? false,
+    }
+    const objToShow = partialClone(obj, options.depth, options.omitKeys, options.replacement)
```

Comment:
should we skip the `partialClone` for the non-web case?

Reply:
I was worried that some of the options would then only apply in certain cases. Ex. if not on web and we pass omitKeys, it does nothing w/o error or warning. But I guess the caller can just wrap it themselves.

Reply:
> Ex. if not on web and we pass omitKeys,

inspect() doesn't really need to support omitKeys though. Unless that is a feature of nodejs inspect().